monai/handlers/mlflow_handler.py

```diff
@@ -222,6 +239,39 @@ def _set_experiment(self):
             raise ValueError(f"Cannot set a deleted experiment '{self.experiment_name}' as the active experiment")
         self.experiment = experiment
 
+    @staticmethod
+    def _get_pandas_dataset_info(pandas_dataset):
+        dataset_name = pandas_dataset.name
+        return {
+            f"{dataset_name}_digest": pandas_dataset.digest,
+            f"{dataset_name}_samples": pandas_dataset.profile["num_rows"],
+        }
+
+    def _log_dataset(self, sample_dict: dict[str, Any], context: str = "nontrain") -> None:
+        if not self.cur_run:
+            raise ValueError("Current Run is not Active to log the dataset")
+
+        # Need to update the self.cur_run to sync the dataset log, otherwise the `inputs` info will be out-of-date.
+        self.cur_run = self.client.get_run(self.cur_run.info.run_id)
+        logged_train_set = [x for x in self.cur_run.inputs.dataset_inputs if "train" == x.dataset.name[: len("train")]]
+        logged_nontrain_set = [
+            x for x in self.cur_run.inputs.dataset_inputs if "nontrain" == x.dataset.name[: len("nontrain")]
+        ]
+        # In case there are more datasets.
+        dataset_cnt = str(len(logged_nontrain_set if context == "nontrain" else logged_train_set))
+        dataset_name = f"{context}_dataset_{dataset_cnt}"
+        sample_df = pandas.DataFrame(sample_dict)
+        dataset = mlflow.data.from_pandas(sample_df, name=dataset_name)
+        exist_dataset_list = list(
+            filter(lambda x: x.dataset.digest == dataset.digest, self.cur_run.inputs.dataset_inputs)
+        )
+
+        if not len(exist_dataset_list):
+            datasets = [mlflow.entities.DatasetInput(dataset._to_mlflow_entity())]
+            self.client.log_inputs(run_id=self.cur_run.info.run_id, datasets=datasets)
```

Comment:
I am concerned about whether we need to support this new feature for now. Seems this feature is still experimental and may change or be removed in a future release without warning, then we will be easily influenced by their release. Or maybe we should fix the version of mlflow to 2.4.0?
https://mlflow.org/docs/latest/python_api/mlflow.client.html?highlight=log_inputs#mlflow.client.MlflowClient.log_inputs

Reply:
Yes, I added this concern in the [issue](https://github.com/Project-MONAI/MONAI/issues/6612#issue-1755085439) when I created it and made the default value of `dataset_log` to be `False`.  I will also update the parameter in the default tracking set to be false. If this is not enough, I think we can pending this PR for now and keep on updating it until mlflow make it a long term API.